puppet/modules/scaleway/manifests/init.pp

```diff
@@ -0,0 +1,26 @@
+# Class to deploy snapshot management for Scaleway hosts
+#
+class scaleway(
+  Hash[String, Hash] $servers       = {},
+  Enum['present', 'absent'] $ensure = present,
+) {
+
+  # To hold the API config
+  file { '/etc/scaleway':
+    ensure => 'directory',
+    owner  => 'root',
+    group  => 'root',
+    mode   => 0700,
+  }
+
+  file { '/usr/bin/scaleway-snapshot':
+    ensure => $ensure,
+    owner  => 'root',
+    group  => 'root',
+    mode   => '0755',
+    source => 'puppet:///modules/scaleway/manage_snapshots.rb',
+  }
+
+  # takes a hash like: { 'server' => { 'api_key' => 'abcde', ord_id => '12345' }, ...}
+  create_resources(scaleway::cron, $servers)
```

Comment:
Technically you could add a third parameter that passes the default value of `$ensure` to each cron resource.

Reply:
See, now you're just *looking* for ways to block me :)

Wouldn't it be better to be able to delete specific cron jobs? Or is this just a default that can be overridden from the $servers hash? Either way, feels like we're splitting hairs now...

Reply:
@ekohl The scare with Scaleway's routers on Friday has reminded me that we should merge this sooner rather than later. Do you have objections to getting this merged and then iterating on it? If so, I'd like to work on this today, so let me know how you'd like to see it move forward

Reply:
You didn't notice that I did ack it - it's just that this would complete it. Personally I would have been to lazy to add an ensure parameter to this class and just hardcoded it to ensure present :)

The third parameter to `create_resources` is indeed the defaults for `$servers` and if that has an ensure parameter that takes precedence.

Reply:
You are correct, I did not notice. Thanks for the ACK, I'll get this rolled out now.